### PR TITLE
Dockerize signsfive-api

### PR DESCRIPTION
**Pull request by meltedspork**
This is partly working. But opened PR for purpose of review. `.env` aren't currently in used since I want to reduce complicated of changes as much as possible.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    working_dir: /home/node/app
+    environment:
+      - NODE_ENV=development
+      - PORT_API=8282
+    volumes:
+      - ./db:/home/node/app/db
+      - ./node_modules:/home/node/app/node_modules
+      - ./scripts:/home/node/app/scripts
+      - ./src:/home/node/app/src
+      - ./.npm:/home/node/.npm
```
Comment (kratsg) [inline docker-compose.yml]: do we need a `.npm` volume? it's in the `.gitignore`

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      - ./.sequelizerc:/home/node/app/.sequelizerc
+      - ./package.json:/home/node/app/package.json
+      - ./package-lock.json:/home/node/app/package-lock.json
+      - ./server.log:/home/node/app/server.log
+    ports:
+      - 8999:8282
```
Comment (kratsg) [inline docker-compose.yml]: how did you decide on the ports?

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    command: npm start
+    networks:
+        signsfive_net:
+          ipv4_address: 172.28.1.2
```
Comment (kratsg) [inline docker-compose.yml]: how did you decide on ip address?

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+  phpmyadmin:
+    container_name: signsfive_phpmyadmin
+    image: phpmyadmin/phpmyadmin
+    depends_on:
+      - database
+    ports:
+      - 9992:80
```
Comment (kratsg) [inline docker-compose.yml]: how did you decide on port?

**Review by kratsg — COMMENTED** [review]
Overall, most of the changes look relatively straightforward...

File: docker-compose.yml (hunk 1 above)
Comment (meltedspork) [reply docker-compose.yml]: Not necessary but if want able to view the log, this is easiest way unless you want to use `docker exec bash`

UPDATE: I just realized. We will want that to only happen in during development mode not production. I will make edit on docker-compose.yml and docker-compose.dev.yml 👍

File: docker-compose.yml (hunk 2 above)
Comment (meltedspork) [reply docker-compose.yml]: This is now in `.env` and only applies to development mode.

File: docker-compose.yml (hunk 3 above)
Comment (meltedspork) [reply docker-compose.yml]: This is now in `.env` and only applies to development mode.

File: docker-compose.yml (hunk 4 above)
Comment (meltedspork) [reply docker-compose.yml]: This is now in `.env` and only applies to development mode.

**Comment by meltedspork** [comment]
I will update documentation about how running in docker and what to do with `.env` on next commit.

**Comment by meltedspork** [comment]
This is now out of WIP and PTAL!

File: README.rst [hunk 5]
```diff
@@ -26,6 +26,8 @@ Next, both ``docker-compose.yml`` and ``docker-compose.dev.yml`` are required, s
 
   npm run dev
 
+NOTE: This might take signsfive_api to restart couple of time due to race condition with database starting up.
```
Comment (kratsg) [inline README.rst]: https://hackernoon.com/preventing-race-conditions-in-docker-781854121ed3

**Review by kratsg — CHANGES_REQUESTED** [review]

**Comment by meltedspork** [comment]
Issue has been open to deal with Docker's race condtions - deafchi/signsfive-api/issues/26

File: README.rst (hunk 5 above)
Comment (meltedspork) [reply README.rst]: deafchi/signsfive-api/issues/26 will try to resolve this race conditions

**Comment by meltedspork** [comment]
I think this is good to be merged in? I made issue to deal with race condtions. Unless it will really break heroku? @kratsg

**Review by kratsg — APPROVED** [review]

**Pull request merged**